backend/src/models/plugins/softDeletePlugin.ts

```diff
@@ -25,6 +25,82 @@ export function softDeletionPlugin(schema: Schema) {
     return await this.save(session)
   }
 
+  schema.statics.deleteOne = async function (
+    filter: Record<string, any>,
+    session?: ClientSession | undefined,
+    user?: string,
+  ) {
+    const update: Record<string, any> = {
+      deleted: true,
+      deletedAt: new Date().toISOString(),
+    }
+    if (user) {
+      update.deletedBy = user
+    }
+
+    const options = session ? { session } : {}
+    return this.updateOne(filter, update, options)
```

Comment:
Is there a reason this is on two lines and not just:

`return this.updateOne(filter, update, { session })`